# Change default port, and ensure endian.h is included when appropriate.
# Specify automake/autoconfig version (automake ignores
#    "AM_INIT_AUTOMAKE([1.15])" in configure.in).
# This change was implemented in-house and is not suitable for upstream use.
#
--- configure.in	2014-04-24 11:05:43.000000000 -0700
+++ configure.in	2016-04-18 16:34:42.645890750 -0700
@@ -11,7 +11,10 @@
 
 # compute $target
 AC_CANONICAL_TARGET
-AM_INIT_AUTOMAKE([foreign 1.6])
+# Make automake-1.15 shut up about "option 'subdir-objects' is disabled"
+AM_INIT_AUTOMAKE([foreign 1.15 subdir-objects])
+# Required by new autoconf 1.15:
+AC_CONFIG_MACRO_DIR([m4])
 
 # Debugging support
 AC_ARG_ENABLE([debug],
@@ -147,7 +150,9 @@
 #
 # The default port that the TCS daemon listens on
 #
-AC_SUBST(TCSD_DEFAULT_PORT, 30003)
+#AC_SUBST(TCSD_DEFAULT_PORT, 30003)
+# 0 designates UNIX Domain socket. For TCP sockets, 30003 is the traditional TCP port.
+AC_SUBST(TCSD_DEFAULT_PORT, 0)
 #
 # The RPC mechanism to build into both libtspi and the tcsd
 #
@@ -355,6 +360,7 @@
 
 AC_C_BIGENDIAN([AC_DEFINE(_BIG_ENDIAN, 1, [big-endian host])])
 AC_CHECK_DECL(htole32, [AC_DEFINE(HTOLE_DEFINED, 1, [htole32 function is available])])
+AC_CHECK_HEADER(endian.h, [AC_DEFINE(HAVE_ENDIAN_H, 1, [endian.h header])])
 AC_CHECK_HEADER(sys/byteorder.h, [AC_DEFINE(HAVE_BYTEORDER_H, 1, [sys/byteorder.h header])])
 AC_CHECK_FUNC(daemon, [ AC_DEFINE(HAVE_DAEMON, 1, [daemon function is available]) ])
  
--- configure	2014-04-24 11:05:49.000000000 -0700
+++ configure	2016-04-18 17:09:18.687255765 -0700
@@ -2544,7 +2544,7 @@
   test "$program_prefix$program_suffix$program_transform_name" = \
     NONENONEs,x,x, &&
   program_prefix=${target_alias}-
-am__api_version='1.13'
+am__api_version='1.15'
 
 # Find a good install program.  We prefer a C program (faster),
 # so one script is as good as another.  But avoid the broken or
--- Makefile.am	2014-04-24 11:05:44.000000000 -0700
+++ Makefile.am	2016-04-18 16:11:59.380522059 -0700
@@ -4,3 +4,5 @@
 	doc/LTC-TSS_LLD_08_r2.pdf \
 	doc/LTC-TSS_LLD_08_r2.sxw \
 	doc/TSS_programming_SNAFUs.txt
+# Suggested by libtoolize:
+ACLOCAL_AMFLAGS = -I m4
